src/runtime_src/core/emulation/hw_em/generic_pcie_hal2/shim.cxx

```diff
@@ -253,13 +263,15 @@ namespace xclhwemhal2 {
     delete[] xmlFile;
     delete[] memTopology;
     delete[] pdi;
+    delete[] emuData;
 
     PRINTENDFUNC;
     return returnValue;
   }
 
```

Comment:
1. Please use early exits coding style; for example you can check for _xclbin_ magic number all in one place in the beginning and exit early if the magic number is incorrect which will remove the indentation
https://llvm.org/docs/CodingStandards.html#use-early-exits-and-continue-to-simplify-code
2. Drop all the legacy xclbin0/xclbin1
3. Use std::unique_ptr<char[]> for char buffer resource management. See dmatest.h for example.

Reply:
Hi Sonal,  updated as per the review comments.